os/kernel/pthread/pthread_keydestroy.c

```diff
@@ -71,35 +71,21 @@
 void pthread_key_destroy(struct pthread_tcb_s *tcb)
 {
 	struct task_group_s *group = tcb->cmn.group;
-	struct pthread_key_s *cur_key;
-	struct pthread_key_s *next_key;
+	pthread_key_t key_index;
 	int destr_count = 0;
 
-	for (cur_key = (struct pthread_key_s *)sq_peek(&tcb->key_list); cur_key; cur_key = next_key) {
+	for (key_index = 0; key_index < PTHREAD_KEYS_MAX; key_index++) {
+		if (group->tg_key[key_index] == KEY_INUSE && tcb->key_data[key_index] != NULL && group->tg_destructor[key_index] != NULL) {
```

Comment:
same as above

Reply:
Since this logic is dependent on key of a particular TCB, We can  optimize this  loop by first checking the " tcb->key_data[key_index] != NULL "  ,   Because group->tg_key[key_index] would hold the keys of other threads as well .

Reply:
Do you mean that ```if (tcb->key_data[key_index] != NULL && group->tg_destructor[key_index] != NULL) {``` is enough?
Don't we need to check validation of key allocated?
My codes means logically that
1. Is that key allocated?
2. Does that key have a data?
3. Does that key have a destructor?

Even if that key is created by other thread, that key is valid on this thread as well.

Reply:
I removed ```group->tg_key[key_index] == KEY_INUSE``` conditional.